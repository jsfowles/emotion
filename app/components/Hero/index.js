import React from 'react';
import PropTypes from 'prop-types';

import {
  HeroWrapper,
  HeroContent,
} from './styles';

import Shadow from '../Shadow';
import Image from '../Image';

const Hero = ({ children, imageUrl }) => (
  <HeroWrapper>
    { children && <HeroContent>
      { children }
    </HeroContent> }

    <Image src={ imageUrl } absolute />
    <Shadow />
  </HeroWrapper>
);

Hero.defaultProps = {
  children: null,
};

Hero.propTypes = {
  children: PropTypes.element,
  imageUrl: PropTypes.string.isRequired,
};

export default Hero;
