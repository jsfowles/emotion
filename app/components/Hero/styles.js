import glamorous from 'glamorous';

import * as c from '../../identity/constants';
import * as s from '../../identity/spacing';
import { HOME_HERO_CONTENT } from '../../identity/zIndex';

export const HeroContent = glamorous.div(
  s.spacing({ padding: 'vertical-xl' }),
  {
    display: 'flex',
    flexDirection: 'column',
    position: 'relative',
    zIndex: HOME_HERO_CONTENT,
  },
);

export const HeroWrapper = glamorous.section(
  s.horizontal(c.PAGE_WRAPPER_SPACING_MOBILE, 'padding'),
  {
    background: c.colors.black001,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: '75vh',
    position: 'relative',
  },
);
