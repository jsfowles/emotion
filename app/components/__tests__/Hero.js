import React from 'react';
import renderer from 'react-test-renderer';

import Hero from '../Hero';

describe('Hero', () => {
  test('renders without children', () => {
    const tree = renderer.create(<Hero imageUrl="http://via.placeholder.com/350x150" />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders with children', () => {
    const tree = renderer.create(
      <Hero imageUrl="http://via.placeholder.com/350x150">
        <h1>Test</h1>
      </Hero>,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
