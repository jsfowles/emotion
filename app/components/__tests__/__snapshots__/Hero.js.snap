// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders with children 1`] = `
<section
  className="css-5xtxfz"
>
  <div
    className="css-1oa2u7k"
  >
    <h1>
      Test
    </h1>
  </div>
  <img
    alt={null}
    className="css-1lgcqub"
    onLoad={null}
    src={null}
    style={null}
  />
  <div
    className="css-yew1ls"
  />
</section>
`;

exports[`Hero renders without children 1`] = `
<section
  className="css-5xtxfz"
>
  <img
    alt={null}
    className="css-1lgcqub"
    onLoad={null}
    src={null}
    style={null}
  />
  <div
    className="css-yew1ls"
  />
</section>
`;
